### ISSUE TITLE
Lazy attributes with fake factories some times fails.

### ISSUE THREAD

**Issue by proofit404** [rbarrois/factory_boy#209]
Hi.
Some times my builds fails with integrity error like this: https://travis-ci.org/proofit404/codeflame/builds/67279773
Here is model definition caused this problem: https://github.com/alex/django-taggit/blob/develop/taggit/models.py#L87-L90
And here is my factory for it: https://github.com/alex/django-taggit/blob/develop/taggit/models.py#L87-L90

I try to wrote it both with `LazyAttribute` and with `Sequence`. Result is pretty much the same. Build will randomly fails with integrity error.

Since there is `Faker` integration in the upcoming `factory_boy` release I think this issue will apply to it too.

Do you know any workaround to this problem?

**Comment by rbarrois** [comment]
@proofit404 I'll need your factory definition, looks like you've mistyped the URL :)

**Comment by proofit404** [comment]
Factory: https://github.com/proofit404/codeflame/blob/564740ee26d586c6ee62f5a848224e9201fc2824/src/tags/tests/factories.py#L10-L17
Model: https://github.com/alex/django-taggit/blob/ac5b7ef403aebcd20850bd809b00825b4c2e49bc/taggit/models.py#L92-L95

**Comment by jeffwidman** [comment]
I'm getting a 404 for your Factory definition. Can you post an updated link?

**Comment by proofit404** [comment]
Sorry, I've deprecate this project. Here is updated link to the [factory](https://bitbucket.org/proofit404/codeflame/src/4a3d87272c56ca94d26fbcb90308b85e57c73184/src/tags/tests/factories.py?at=master&fileviewer=file-view-default#factories.py-10:17).

**Comment by jeffwidman** [comment]
Thanks, but I'm getting a Bitbucket login screen which means it's not a public repo... Can you make it public?

**Comment by proofit404** [comment]
```python
class TagFactory(factory.DjangoModelFactory):

    name = factory.Sequence(lambda num: fake.word())
    slug = factory.LazyAttribute(lambda obj: obj.name)

    class Meta:

        model = 'taggit.Tag'
```

**Comment by rbarrois** [comment]
@proofit404 If I read your factory correctly, you're using a random value in a unique field.

Depending on the values generated by your random generator, this can produce the same value twice, thus leading to the ``IntegrityError`` you're encountering.

You could probably fix this by replacing your sequence with ``name = factory.Sequence(lambda n: '%s %d' % (fake.word(), n))``.

**Comment by proofit404** [comment]
@rbarrois Actually I already append sequence number to fake.word result. I create this issue because of new build-in faker integration which can cause the same error.

**Comment by jeffwidman** [comment]
I suspect this is because the `fake.word()` generator pulls values from the Lorem corpus. If so, the Lorem corpus has multiple duplicate words, and so isn't actually unique, despite what the Faker docs imply. 

I got bit by something similar and submitted a PR to them over a year ago: https://github.com/joke2k/faker/pull/177 I'd appreciate if you leave a note over on that PR mentioning that you got bit by the issue as well.

Closing as this isn't an issue with Factory Boy.
